chia/cmds/param_types.py

```diff
@@ -0,0 +1,225 @@
+                raise ValueError("Amount must be a uint64 if mojos flag is set.")
+            return self.amount
+        if not isinstance(self.amount, Decimal):
+            raise ValueError("Amount must be a Decimal if mojos flag is not set.")
+        return uint64(self.amount * mojo_per_unit)
+
+
+class AmountParamType(click.ParamType):
+    """
+    A Click parameter type for TX / wallet amounts for both XCH and CAT, and of course mojos.
+    """
+
+    name: str = CliAmount.__name__  # output type
+
+    def convert(self, value: Any, param: Optional[click.Parameter], ctx: Optional[click.Context]) -> CliAmount:
+        # suggested by click, but being left in as mojos flag makes default misrepresentation less likely.
+        if isinstance(value, CliAmount):
+            return value
+        if not isinstance(value, str):
+            self.fail("Invalid Type, amount must be string or CliAmount.", param, ctx)
+        mojos = False  # TODO: Add unit logic
+        if mojos:
+            m_value = validate_uint64(value, self.fail, param, ctx)
+            return CliAmount(mojos=True, amount=m_value)
+        d_value = validate_decimal_xch(value, self.fail, param, ctx)
+        return CliAmount(mojos=False, amount=d_value)
+
+
+@dataclass(frozen=True)
+class CliAddress:
+    """
+    A dataclass for the cli, with the address type and puzzle hash.
+    """
+
+    puzzle_hash: bytes32
+    original_address: str
+    address_type: AddressType
+
+    def validate_address_type(self, address_type: AddressType) -> str:
+        if self.address_type is not address_type:
+            raise ValueError(f"Address must be of type {address_type}")
+        return self.original_address
+
+    def validate_address_type_get_ph(self, address_type: AddressType) -> bytes32:
+        if self.address_type is not address_type:
+            raise ValueError(f"Address must be of type {address_type}")
+        return self.puzzle_hash
+
+
+class AddressParamType(click.ParamType):
+    """
+    A Click parameter type for bech32m encoded addresses, it gives a class with the address type and puzzle hash.
+    """
+
+    name: str = CliAddress.__name__  # output type
+
+    def convert(self, value: Any, param: Optional[click.Parameter], ctx: Optional[click.Context]) -> CliAddress:
+        # suggested by click, but not really used so removed to make unexpected types more obvious.
+        # if isinstance(value, CliAddress):
+        #    return value
+        if not isinstance(value, str):
+            self.fail("Invalid Type, address must be string.", param, ctx)
+        try:
+            hrp, b32data = bech32_decode(value)
+            if hrp in ["xch", "txch"]:  # I hate having to load the config here
+                expected_prefix = ctx.obj.get("expected_prefix") if ctx else None
+                if expected_prefix is None:
+                    from chia.util.config import load_config
+                    from chia.util.default_root import DEFAULT_ROOT_PATH
```

Comment:
i just did that to reduce load time if it isnt needed but im not sure if that matters.

Reply:
The intent is certainly valid and I think it would be good for us to improve that.  In practice, I think we already import every everything already and I would rather deal with this consistently with a plan than ad-hoc here and there sometimes.  For example, we could look at configuring a lazy import strategy that doesn't even involve changing each import site.

Reply:
ok so like move it, got it